src/libsync/discovery.cpp

```diff
@@ -1002,17 +1002,18 @@ void ProcessDirectoryJob::processFileAnalyzeLocalInfo(
             serverOriginalPath = chopVirtualFileSuffix(serverOriginalPath);
         }
         auto job = new RequestEtagJob(_discoveryData->_account, _discoveryData->_baseUrl, serverOriginalPath, this);
-        connect(job, &RequestEtagJob::finishedWithResult, this,
-            [recurseQueryServer, path = path, postProcessLocalNew, processRename, base, item, originalPath, this](const HttpResult<QByteArray> &etag) {
-                if (!etag || (etag.get() != base._etag && !item->isDirectory()) || _discoveryData->isRenamed(originalPath)) {
+        connect(job, &RequestEtagJob::finishedSignal, this,
+            [job, recurseQueryServer, path = path, postProcessLocalNew, processRename, base, item, originalPath, this] {
+                // TODO: discuss: the original code emitted error types for http codes != 207, but is an empty value valid? see line 1016
+                if (job->httpStatusCode() != 207 || (job->etag() != base._etag && !item->isDirectory()) || _discoveryData->isRenamed(originalPath)) {
```

Comment:
I don't think empty values make any sense regarding the spec's wording, see https://www.rfc-editor.org/rfc/rfc7232#section-2.3. Not sure how our own API handles this, though.

There seem to be WebDAV servers out there that set a valid ETag for 207 responses...

"No ETag set" might be valid here, but ultimately it depends on the server. I can't really tell how it handles these cases...

By the way, maybe treat `Etag: ` (no value) as errors, since this would make absolutely no sense.